src/Data/Container/Indexed/Relation/Binary/Equality/Setoid.agda

```diff
@@ -0,0 +1,59 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Equality over indexed container extensions parametrised by a setoid
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Relation.Binary using (Setoid)
+
+module Data.Container.Indexed.Relation.Binary.Equality.Setoid
+  {ℓⁱ ℓᶜ ℓᵉ} {I : Set ℓⁱ} (S : I → Setoid ℓᶜ ℓᵉ)
+  where
+
+open import Function
+open import Level using (Level; _⊔_; suc)
+open import Relation.Binary
+
+open import Data.Container.Indexed.Core
+open import Data.Container.Indexed.Relation.Binary.Pointwise
+import Data.Container.Indexed.Relation.Binary.Pointwise.Properties as Pw
+
+open Setoid using (Carrier) renaming (_≈_ to Equality)
+
+private variable
+  ℓˢ ℓᵖ : Level
+  O : Set _
+
+------------------------------------------------------------------------
+-- Definition of equality
+
+module _ (C : Container I O ℓˢ ℓᵖ) (o : O) where
+
+  Eq : Rel (⟦ C ⟧ (Carrier ∘ S) o) (ℓᵉ ⊔ ℓˢ ⊔ ℓᵖ)
```

Comment:
If you do lift it the above arguments to be module arguments, this could then become infix?